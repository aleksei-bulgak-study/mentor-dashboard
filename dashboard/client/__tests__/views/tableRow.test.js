import React from 'react';
import { shallow } from 'enzyme';
import TableRow from '../../src/views/tableRow';

describe('TableRow', () => {
  it('default snapshot', () => {
    // given
    const props = {
      task: { name: 'test', status: 'Checking', link: 'https://example.com/test' },
      students: ['student1', 'student2'],
      results: [
        {
          github: 'https://github.com/student1',
          tasks: {
            test: { mark: 100, pr: 'https://example.com/task' },
            test2: { mark: 30, pr: 'https://example.com/task2' },
          },
        },
      ],
    };
    // when
    const wrapper = shallow(<TableRow {...props} />);
    // then
    expect(wrapper).toMatchSnapshot();
  });
});
