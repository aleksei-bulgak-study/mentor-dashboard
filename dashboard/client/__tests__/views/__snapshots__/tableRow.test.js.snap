// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableRow default snapshot 1`] = `
<tr>
  <td
    key="test"
  >
    <a
      href="https://example.com/test"
      rel="noopener noreferrer"
      target="_blank"
    >
      test
    </a>
  </td>
  <TaskResultsContainer
    results={
      Array [
        Object {
          "github": "https://github.com/student1",
          "tasks": Object {
            "test": Object {
              "mark": 100,
              "pr": "https://example.com/task",
            },
            "test2": Object {
              "mark": 30,
              "pr": "https://example.com/task2",
            },
          },
        },
      ]
    }
    students={
      Array [
        "student1",
        "student2",
      ]
    }
    task={
      Object {
        "link": "https://example.com/test",
        "name": "test",
        "status": "Checking",
      }
    }
  />
</tr>
`;
